submariner/pluggable-network-plugin-and-ovn.md

```diff
@@ -0,0 +1,50 @@
+## Add support for OVN with pluggable network-plugin support
+
+[submariner#778](https://github.com/submariner-io/submariner/issues/778)
+
+## Summary
+
+Today submariner supports working with existing network-plugins which are based
+on the kubeproxy and iptables implementations. Any network-plugin outside of this
+case won't work, OVN-Kubernetes is an example.
+
+This is a proposal to support pluggingable network drivers, starting by OVN-kubernetes.
+
+## Proposal
+
+Re-architect submariner-routeagent in a way that different network-plugin implementations
+can be provided. Those drivers would be notified of different transitions on a node:
+
+* Transition of a gateaway node in your cluster
+* Connection of a new cluster (and it's CIDR details)
+* Disconnection of existing clusters (and it's CIDR details)
+* ... any other event necessary -TBD-
```

Comment:
Initialization events to handle any CNI Specific tweaks, creating necessary chains etc
Notification when new route-agent pods are scheduled in the local cluster (needed for vxlan based drivers)

Reply:
+1, I'll add that